rio_tiler/mosaic/reader.py

```diff
@@ -59,6 +59,9 @@ def reader(asset: str, *args, **kwargs) -> Tuple[numpy.ndarray, numpy.ndarray]:
         Return (tile, mask) data and list of assets used.
 
     """
+    if issubclass(pixel_selection, MosaicMethodBase):
```

Comment:
If think it will raise an error if pixel_selection is not a class 
```
import inspect
if inspect.isclass(pixel_selection) and issubclass(pixel_selection, MosaicMethodBase):
    pixel_selection = pixel_selection()
```